src/launchpad/analyzers/android.py

```diff
@@ -82,9 +83,11 @@ def analyze(self, artifact: AndroidArtifact) -> AndroidAnalysisResults:
                 file_analysis=file_analysis,
                 treemap=treemap,
                 binary_analysis=[],
+                artifact=artifact,
```

Comment:
Let me know if there’s a better way to do this. We need to open the files from the artifact in order to get access to the images.

Reply:
I don't think this is the pattern we want to follow. Both treemap/insights shouldn't know about the artifact. The analyzer is responsible for taking the artifact and doing the "analysis" on it, then generating the treemap/insights from that analysis. I think a separate function in `android.py` here to find all images and check if they can be optimized, then return the list of optimizable images that the InsightsInput would take would be better.

cc @trevor-e as we've been discussing this a bit offline